docs/extensions/garden-api-access.md

```diff
@@ -127,8 +127,36 @@ With this, extensions are restricted to work with objects in the garden cluster
 Note that if the plugins are not enabled, extension clients are only granted read access to global resources like `CloudProfiles` (this is granted to all authenticated users).
 There are a few exceptions to the granted permissions as documented [here](../deployment/gardenlet_api_access.md#rule-exceptions-for-extension-clients).
 
+### Additional Permissions
```

Comment:
I don't think that this was the exact intention of the issue which says:

> use a dedicated service account token as well as customizable permissions because the ones provided for the extension are either too restrictive or too open.

Additional permissions solve the "too restrictive" case but doesn't cover the "too open" one. While being in clarification with @nickytd, I understood that the goal is not to fully inherit existing permissions to the new `ServiceAccount`. For security reasons, it should only have the ones specific in the `ClusterRole`.

Is this already supported? IIRC, the Seed Authorizer relies on a `extension` prefix ...

Reply:
Reducing the permissions is currently not possible - this PR is only about extending them. I'm also unsure if the current approach would be a good way to implement it (I didn't think about this use case yet, but I guess it will not work well together with the seed authorizer (e.g., if the `ClusterRole` allowed to delete shoots, then the extension could just delete all shoots, no matter whether they relate to its seed or not, since RBAC would already allow the request (and seed authorizer won't be asked about it))). This is probably not what we want to achieve.

Reply:
That's already the case today, isn't it?
The following `ClusterRole` will allow the `ServiceAccount`(s) to delete random shoots with the current implementation:

```
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRole
metadata:
  annotations:
    authorization.extensions.gardener.cloud/bind-to: provider-local
  labels:
    authorization.extensions.gardener.cloud/additional-permissions: "true"
  name: extension-shoot-mutation
rules:
- apiGroups:
  - core.gardener.cloud
  resources:
  - shoots
  verbs:
  - '*'
```

Reply:
True :) All these downsides are reasons for not having invested into these features (changing extension permissions) earlier. I also doubt that we can find a perfect solution just by using RBAC (otherwise, we wouldn't have implemented the seed authorizer for the gardenlet permissions).
FMPOV, extending the permissions as proposed in this PR is acceptable (with the risk of human operators to make mistake like [this](https://github.com/gardener/gardener/pull/9066#discussion_r1464730794) (though, this is already the case today, i.e., human operators can easily grant too wide permissions via RBAC which bypass the seed authorizer)). Reducing the permissions is more complicated and I would continue to refrain from looking into it for now (especially in the light that all extensions run in the same seed and can easily "steal" the gardenlet kubeconfig, so the overall security improvement is rather limited/minimal).

An alternative to this PR is to do what you apparently originally had in mind, i.e., allow human operators to define a `ClusterRole` for the extensions and let them completely bypass the seed authorizer. TBH, I'm not too much in favor of a concept like this which relies on "intentionally deactivating the seed authorizer" (while [this](https://github.com/gardener/gardener/pull/9066#discussion_r1464730794) rather looks like a mistake/error).
I understand that this would be sufficient/suitable for @nickytd's use case, but that's probably a pretty specific one. My expectation would be that a more common scenario is to add additional permissions to extensions for resources not covered by the seed authorizer (e.g., the `SubjectAccessReview`s).

TL;DR: I'm also not fully convinced by this PR, but I think that it is a good compromise without being a solution that's "too special" for this single scenario.

Reply:
>  Reducing the permissions is more complicated and I would continue to refrain from looking into it for now

Is it really that complex? If we choose to follow the name matching (https://github.com/gardener/gardener/pull/9066#discussion_r1464598490), it'll come out of the box.